### PR TITLE
feat/zep vectorstore

### PR DESCRIPTION
**Pull request by danielchalef**
# Description

Zep Vector Store implementation, notebook example, and docs.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

# How Has This Been Tested?

Notebook pretty much tests end-to-end. Tested against Zep 0.9. Note that I need to push a patch-level bugfix to Zep in order to make the metadata filtering work. Will happen tonight.

- [ ] Added new unit/integration tests
- [X] Added new notebook (that tests end-to-end)
- [X] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by Disiok — COMMENTED** [review]
This looks great! Thanks for the contribution.

Small ask: could you add an entry to the table in `docs/core_modules/data_modules/storage/vector_stores.md`  as well? Mostly to document which subset of features are supported.

**Comment by danielchalef** [comment]
> This looks great! Thanks for the contribution.
> 
> Small ask: could you add an entry to the table in `docs/core_modules/data_modules/storage/vector_stores.md` as well? Mostly to document which subset of features are supported.

@Disiok Done!

**Comment by danielchalef** [comment]
/-: My IDE reformatted the markdown table. Sorry for the large diff...

**Review by logan-markewich — APPROVED** [review]
Awesome, looks good to me! Will merge shortly

File: llama_index/vector_stores/zep.py [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+
+    def _parse_query_result(
+        self, results: List["ZepDocument"]
+    ) -> VectorStoreQueryResult:
+        similarities: List[float] = []
+        ids: List[str] = []
+        nodes: List[TextNode] = []
+
+        for d in results:
+            node = TextNode(text=d.content, metadata=d.metadata)
```
Comment (logan-markewich) [inline llama_index/vector_stores/zep.py:230]: Actually, this is not quite correct

We should use the metadata dict to node util function here. I can patch that in

Comment (danielchalef) [reply llama_index/vector_stores/zep.py]: Thank you!

Comment (logan-markewich) [reply llama_index/vector_stores/zep.py]: Fixed!

**Pull request merged**